lib/elasticsearchHandler.js

```diff
@@ -60,18 +143,42 @@ ElasticsearchStore.prototype.initialise = function(resourceConfig) {
     self._db = client;
   }
   self.resourceConfig = resourceConfig;
-  if (!self._db) {
-    return elasticsearch.resourcesToInitialise.push(resourceConfig);
-  }
   self.ready = true;
 };
 
+ElasticsearchStore.prototype.populate = function(callback) {
+  var self = this;
+  self._db.indices.delete({ index: "jsonapi" }, function(err) {
+    if (err) console.log("Error dropping index?", err.message);
+
+    self._db.indices.create({ index: "jsonapi" }, function(err1) {
+      if (err1) console.log("Error creating index?", err1);
+
+      var mappingRequest = {
+        index: "jsonapi",
+        type: self.resourceConfig.resource,
+        body: self._generateMappingFor(self.resourceConfig)
+      };
+      self._db.indices.putMapping(mappingRequest, function(err2) {
```

Comment:
We could, in all honesty this function is used exclusively by the test runner to setup enough to enable us to query, sort and pagination in the way the tests expect. It's not ideal in that it'll attempt to delete the `jsonapi` index once for every elasticsearch handler present in the test suite, but it does at least function...